### PR TITLE
Introduce a way to disable JAX-RS resources at runtime

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java [hunk 1]
```diff
@@ -589,6 +589,17 @@ private boolean hasAnnotation(MethodInfo method, short paramPosition, DotName an
                         public Status isJava19OrHigher() {
                             return result;
                         }
+                    })
+                    .setIsDisabledCreator(new Function<>() {
+                        @Override
+                        public Supplier<Boolean> apply(ClassInfo classInfo) {
+                            //TODO: what annotation should be used? @IfBuildProperty, @LookupIfProperty aren't semantically the same...
```
Comment (ivansenic) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java:596]: what about a property similar for retries:

`io.some.ClassName/methoName/Retry/max-retries=0`

Could something like this work here.. Not sure if this is preferred though..

`io.some.ResourceClass/disabled=true`

Not sure how this works with ENV vars.. Maybe it s a stupid idea..

Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java:596]: The reason I usually don't like to go down the properties route is that it's hard for users to write them as the IDE will not help you at all.

Comment (mkouba) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java:596]: I think that we should introduce a new annotation; e.g. something like `@EndpointDisabled`.

**Comment by geoand** [comment]
Should be good to go now

**Review by mkouba — APPROVED** [review]
Looks good.

**Comment by tatu-at-datastax** [comment]
Excellent!

And from milestone, it would seem this could end in `2.16.0` release? That would be great.

**Comment by geoand** [comment]
Exactly, that's the plan

**Pull request merged**